File: Codes in Python3/Graph/No. of enclaves(using BFS).py
class Solution:
    def numEnclaves(self, grid: List[List[int]]) -> int:
        m=len(grid)
        n=len(grid[0])
        vis=[[0]*n for _ in range(m)]
        q=[]
        count=0
        for i in range(m):
            for j in range(n):
                if i==0 or j==0 or i==m-1 or j==n-1:
                    if vis[i][j]==0 and grid[i][j]==1:
                        q.append((i,j))
                        vis[i][j]=1
        while q:
            row,col=q.pop(0)
            delrow=[-1,0,1,0]
            delcol=[0,1,0,-1]
            for i in range(4):
                nrow=row+delrow[i]
                ncol=col+delcol[i]
                if nrow>=0 and nrow<m and ncol>=0 and ncol<n and vis[nrow][ncol]==0 and grid[nrow][ncol]==1:
                    q.append((nrow,ncol))
                    vis[nrow][ncol]=1

        for i in range(m):
            for j in range(n):
                if vis[i][j]==0 and grid[i][j]==1:
                    count+=1
        return count

File: Codes in C++/Graph/No. of enclaves(using BFS).cpp
class Solution {
public:
    int numEnclaves(vector<vector<int>>& grid) {
    int m=grid.size();
    int n=grid[0].size(); 
    vector<vector<int>> vis(m,vector<int>(n,0));
    queue<pair<int,int>> q;
    for(int i=0;i<m;i++)
    {
        for(int j=0;j<n;j++)
        {
            if(i==0||j==0||i==(m-1)||j==(n-1))
            {
                if(vis[i][j]==0 && grid[i][j]==1)
                   { q.push({i,j});
                    vis[i][j]=1;
                   }
            }
    }   
    }
    int count=0;
    while(!q.empty())
    {
        int row=q.front().first;
        int col=q.front().second;
        q.pop();
        int delrow[]={-1,0,1,0};
        int delcol[]={0,1,0,-1};
        for(int i=0;i<4;i++)
        {
            int nrow=row+delrow[i];
            int ncol=col+delcol[i];
            if(nrow>=0 && nrow<m && ncol>=0 && ncol<n&&
            vis[nrow][ncol]==0 && grid[nrow][ncol]==1)
            {
                q.push({nrow,ncol});
                vis[nrow][ncol]=1;
            }
        }
    }

    for(int i=0;i<m;i++){
        for(int j=0;j<n;j++)
        {
            if(vis[i][j]==0 && grid[i][j]==1)
                count+=1;
        }   
    }
    return count;
    }
};